src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
+
 export interface Figure {
+  color: Color,
+  shape: Shape,
 
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c].sort((x, y) => y - x);
 
-export class Circle {
+    if ((sides[0] >= sides[1] + sides[2])
+        || a < 1 || a <= 1) {
```

Comment:
these two evaluations seem contradictory
if a = 1: 
```
  || a < 1 // false
  || a <= 1 // true
```
based on your evaluation , an Error will be thrown, yet it shouldn't